pkg/api/handlers/compat/events.go

```diff
@@ -63,16 +64,25 @@ func GetEvents(w http.ResponseWriter, r *http.Request) {
 		errorChannel <- runtime.Events(r.Context(), readOpts)
 	}()
 
-	flush := func() {}
+	// Use a larger buffer to aid some historical docker clients that expect
+	// an event to fit in every read() call
+	o := bufio.NewWriterSize(w, 8192)
```

Comment:
Docker seems to use WriteFlusher https://github.com/moby/moby/blob/master/api/server/router/system/system_routes.go#L244-L246.

Should we do the same?  It's available in `containers/storege/pkg/ioutils`.

Reply:
So that's the funny thing. Flushing on every write like that should increase the odds of fragmentation. I haven't looked too deeply, but theres a good chance the batched per event behavior is just timing luck (lots of small writes right next to each other that happen to get read together and a pause between events from the added locking that by luck reduces the chance of reading past/across events. Changing to a larger buffer should be more reliable, well as reliable as we can be with this brittle pattern. There could be other factors going on, though.